oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java

```diff
@@ -430,6 +444,18 @@ public Builder jwkSetUri(String jwkSetUri) {
 			return this;
 		}
 
+		/**
+		 * Sets the metadata describing the provider's configuration.
+		 *
+		 * @since 5.1
+		 * @param configurationMetadata the metadata describing the provider's configuration
+		 * @return the {@link Builder}
+		 */
+		public Builder providerConfigurationMetadata(Map<String, Object> configurationMetadata) {
+			this.configurationMetadata = configurationMetadata;
```

Comment:
In combination with making `configurationMetadata` empty, we should also prevent this from be null